crucible/src/Lang/Crucible/Simulator/RegMap.hs

```diff
@@ -144,8 +150,9 @@ abortBranchForType s iTypes p =
     IntrinsicRepr nm ctx ->
        case MapF.lookup nm iTypes of
          Just IntrinsicMuxFn -> abortBranchIntrinsic s iTypes nm ctx
-         Nothing ->
-           fail $ unwords ["Unknown intrinsic type:", show nm]
+         Nothing -> \_ ->
+           addFailedAssertion s
+              $ Unsupported $ unwords ["Unknown intrinsic type:", show nm]
```

Comment:
This should probably be a panic.  Missing an intrinsic mux fn indicates that somebody forgot to pipe in the necessary runtime information when using the library.